ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveInsertExchange4JoinRule.java

```diff
@@ -118,7 +127,7 @@ public void onMatch(RelOptRuleCall call) {
         for (int pos : joinLeafPredInfo.getProjsJoinKeysInChildSchema(i)) {
           if (!joinKeyPositions.contains(pos)) {
             joinKeyPositions.add(pos);
-            collationListBuilder.add(new RelFieldCollation(pos));
+            collationListBuilder.add(new RelFieldCollation(pos, RelFieldCollation.Direction.ASCENDING, defaultAscNullDirection));
```

Comment:
Do we always want the direction to be `ASCENDING`?

Reply:
The reason why we need these sort exchange operators is to get the data coming from join branches to be sorted because we want to perform a sort merge join.
I think Ascending is the default but it could also work with descending. I think it has an affect only on queries with limit but without order/sort by. However it doesn't matter which n rows do we pick in those cases.